src/services/upload/metadataService.ts

```diff
@@ -149,3 +159,30 @@ export async function parseMetadataJSON(
         // ignore
     }
 }
+
+// tries to extract date from file name if available else returns null
+export function extractDateFromFileName(filename: string): number {
+    filename = filename.trim();
+    let parsedDate: Date;
+    if (filename.startsWith('IMG-') || filename.startsWith('VID-')) {
+        // Whatsapp media files
+        parsedDate = parseDateFromFusedDateString(filename.split('-')[1]);
+    } else if (filename.startsWith('Screenshot_')) {
+        // Screenshots on droid
+        parsedDate = parseDateFromFusedDateString(
+            filename.replaceAll('Screenshot_', '')
+        );
+    } else if (filename.startsWith('signal-')) {
+        // signal images
+        const dateString = convertSignalNameToFusedDateString(filename);
+        parsedDate = parseDateFromFusedDateString(dateString);
+    } else {
+        parsedDate = tryToParseDateTime(filename);
+    }
+    return getUnixTimeInMicroSeconds(parsedDate);
+}
+
+function convertSignalNameToFusedDateString(filename: string) {
+    const dateStringParts = filename.split('-');
+    return `${dateStringParts[1]}${dateStringParts[2]}${dateStringParts[3]}-${dateStringParts[4]}`;
```

Comment:
what is there's not enough `-` in the name?

Reply:
so this is for signal files and they have the format of `YYYY-MM-DD-HHMMSS` hence we would always have the required number of dashes

Reply:
Cool. I guess just having this entire `extractDateFromFileName` inside try catch will help me sleep better :D